registry/remote/auth/client.go

```diff
@@ -57,9 +57,9 @@ var defaultClientID = "oras-go"
 // CredentialFunc represents a function that resolves the credential for the
 // given registry (i.e. host:port).
 //
-// `EmptyCredential` is a valid return value and should not be considered as
+// [EmptyCredential] is a valid return value and should not be considered as
 // an error.
-type CredentialFunc func(ctx context.Context, registry string) (Credential, error)
+type CredentialFunc func(ctx context.Context, target string) (Credential, error)
```

Comment:
`target` is ambiguous. It is indeed `hostname` and `port`.

How about `hostport` as [net.SplitHostPort()](https://pkg.go.dev/net#SplitHostPort)?

Reply:
That works.